proposals/0197-chilli-peppers.md

```diff
@@ -0,0 +1,147 @@
+ 
+Block Chili Pepper Limit - The maximum number of Chili Peppers that can be requested in a single block.
+ 
+Block Chili Pepper Clock - a cumulative measure of the total Chili Peppers requested since the genesis of the blockchain. At the beginning of each block, the Block State Unit Clock is updated to reflect the cumulative total of Chili Peppers consumed up to that point, ensuring a monotonically increasing record of state consumption.
+ 
+Account Chili Pepper Clock Timestamp - each account in the "hot" state needs to keep track of the value of the Block Chili Pepper Clock the last time it was accessed (read or written to). This allows determination of which accounts are hot and which are cold.
+ 
+Hot Cache Size - A new predefined constant, termed "Hot Cache Size" corresponds approximately to the size of the hot state supported
+by the validator network. An account is designated as hot if it was last
+accessed within this threshold of the current Block Chili Pepper Clock
+(Account Chili Pepper Clock Timestamp > Block Chili Pepper Clock - Hot Cache Size), otherwise it is designated
+as cold.
+ 
+ 
+## Detailed Design
+ 
+### Integration with ComputeBudget Program
+ 
+To facilitate the utilization of Chili Peppers, every transaction on the Solana network will be required to use the ComputeBudget program to request the maximum number of Chili Peppers they require. Alongside existing functionalities, a new instruction will be introduced to specify the requested State Units for the transaction. This ensures that developers have the flexibility to request resources based on the anticipated needs of their transactions, within the constraints of the block's Chili Peppers capacity.
+The new ComputeBudgetInstruction will be as follows:
+ 
+```rust
+discriminant: 5
+ComputeBudgetInstruction::SetChiliPepperLimit(u32)
+```
+ 
+The 32-bit unsigned integer in the instruction indicates the number of Chili Peppers requested for the transaction.
+ 
+### Block Chili Pepper Clock
+ 
+The Block Chili Pepper Clock serves as a cumulative measure of the total Chili Peppers requested since the genesis of the blockchain. At the beginning of each block, the Block State Unit Clock is updated to reflect the cumulative total of Chili Peppers requested up to that point, ensuring a monotonically increasing record of state consumption.
+ 
+Computing the block Chili Pepper clock:
+ 
+```python
+block_chili_pepper_clock = prev_block_chili_pepper_clock + sum(txn.requested_chili_peppers for txn in block.txns)
+```
+ 
+Implemented as a 64-bit unsigned integer (uint64), this clock is updated at the beginning of every block to reflect the total Chili Peppers requested since the chain's genesis. This monotonically increasing value is stored in a dedicated system variable (sysvar), ensuring that it remains accessible and immutable throughout the blockchain's operation.
```

Comment:
begining or end?

Without actually looking up the account, how would a block know how many chili peper are actually used?

Reply:
accessing the meta-data for an account needs to be accounted for via chili-peppers as well?